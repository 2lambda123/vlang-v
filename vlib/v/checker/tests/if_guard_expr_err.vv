fn foo() ! {
	return error("error")
}

fn main() {
	a := if r := foo() {
		true
	} else {
		false
	}
	
	println(a)
}
