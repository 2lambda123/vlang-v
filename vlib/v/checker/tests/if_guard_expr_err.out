vlib/v/checker/tests/if_guard_expr_err.vv:6:10: warning: unused variable: `r`
    4 |
    5 | fn main() {
    6 |     a := if r := foo() {
      |             ^
    7 |         true
    8 |     } else {
vlib/v/checker/tests/if_guard_expr_err.vv:6:7: error: if guard expects non-propagate option or result
    4 |
    5 | fn main() {
    6 |     a := if r := foo() {
      |          ~~~~~~~~~~~~~
    7 |         true
    8 |     } else {
